testsuite/pytest_extensions.py

```diff
@@ -137,3 +161,146 @@ def test_CIMClass_equal(
 
             ret = None  # Debugging hint
     return ret
+
+
+def test_function_new(test_func):
+    """
+    A decorator for test functions that simplifies the test function by
+    handling a number of things:
+
+    * Skipping the test if the `condition` item in the testcase is `False`,
+    * Invoking the Python debugger if the `condition` item in the testcase is
+      the string "pdb",
+    * Capturing and validating any warnings issued by the test function,
+      if the `exp_warn_types` item in the testcase is set,
+    * Catching and validating any exceptions raised by the test function,
+      if the `exp_exc_types` item in the testcase is set.
+
+    This is a signature-changing decorator.
+
+    Parameters of the wrapper function:
+
+    * desc (string): Short testcase description.
+
+    * kwargs (dict): Keyword arguments for the test function.
+
+    * exp_exc_types (Exception or list of Exception): Expected exception types,
+      or `None` if no exceptions are expected.
+
+    * exp_warn_types (Warning or list of Warning): Expected warning types,
+      or `None` if no warnings are expected.
+
+    * condition (bool or 'pdb'): Boolean condition for running the testcase.
+      If it evaluates to `bool(False)`, the testcase will be skipped.
+      If it evaluates to `bool(True)`, the testcase will be run.
+      The string value 'pdb' will cause the Python pdb debugger to be entered
+      before calling the test function.
+
+    Parameters of the test function that is decorated:
+
+    * testcase (testcase_tuple): The testcase, as a named tuple.
+
+    * **kwargs: Keyword arguments for the test function.
+
+    Notes:
+
+    * Using this decorator together with the `pytest.mark.parametrize`
```

Comment:
How about when applying this decorator with param... this decorator must be inserted before the param... decorator

Reply:
Should be inserted after the param, not before.

Reply:
DONE.